### PR TITLE
feat: add support of typescript output format

### PR DESCRIPTION
**Pull request by khitrind**

### CODE REVIEW

File: messgen/ts_generator.py [hunk 1]
```diff
@@ -12,150 +16,166 @@
...
-    f_type = t[0].lower() + t[1:]
+class TypeScriptGenerator:
+    def __init__(self, protos: Protocols, options: dict):
+        self.protos = protos
+        self.options = options
+        self.generated_types = {}
+        self.imports = set()
+        self.code_lines = []
+
+    def to_camel_case(self, s):
```
Comment (lachem) [inline messgen/ts_generator.py:32]: How about 
 
```suggestion
    def _to_camel_case(self, s: str) -> str:
```

* leading `_` means private by convention
* the type hints are useful for correctness checks with mypy

File: messgen/ts_generator.py [hunk 2]
```diff
@@ -12,150 +16,166 @@
...
+            fields = type_def.get("fields") or []
+            for field in fields:
+                field_comment = field.get("comment", "")
+                field_name = field["name"]
+                field_type = field["type"]
+                ts_type = self.get_ts_type(field_type, proto_name)
+                if field_comment:
+                    self.code_lines.append(f"  /** {field_comment} */")
+                self.code_lines.append(f"  {field_name}: {ts_type};")
+            self.code_lines.append("}\n")
```
Comment (lachem) [inline messgen/ts_generator.py:117]: AFAIK, the convention for adding new lines is to use `append("")` like here: https://github.com/microavia/messgen/blob/v1-dispatch/messgen/cpp_generator.py#L126

```suggestion
            self.code_lines.append("}")
            self.code_lines.append("")
```

File: messgen/ts_generator.py [hunk 3]
```diff
@@ -12,150 +16,166 @@
...
+                value_comment = value.get("comment", "")
+                if value_comment:
+                    self.code_lines.append(f"  /** {value_comment} */")
+                self.code_lines.append(f"  {name} = {value_value},")
+            self.code_lines.append("}\n")
+        else:
+            pass  
+
+    def get_ts_type(self, field_type, current_proto_name):
+        # Обработка массивов, карт и вложенных типов
```
Comment (lachem) [inline messgen/ts_generator.py:133]: I think it would be beneficial to use English for comments as it would be understandable for a wider audience.

Comment (khitrind) [reply messgen/ts_generator.py:133]: yeah sure, sorry, i thought i opened a pull request in draft, there are a couple of things i wanted to fix, but thanks a lot for the review

**Comment by khitrind** [comment]
I will reopen PR not from a fork

**Pull request closed without merging**